fotoapparat/src/main/java/io/fotoapparat/hardware/v1/Camera1.java

```diff
@@ -218,6 +220,18 @@ public void updateParameters(Parameters parameters) {
         parametersOperator().updateParameters(parameters);
 
         cachedZoomParameters = null;
+        currentParameters = parameters;
```

Comment:
Those parameters will contain only those values which you are updating. For example, if you are updating only flash mode it will not contain picture size.

Actually, the whole thing can be done a bit simpler, without the need for a latch. You can use `camera.getParameters()` to get current parameters of the camera and then map them to our `Parameters` object (by adding this functionality to `ParametersConverter`).